app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt

```diff
@@ -0,0 +1,251 @@
+              subject = Subject.PHYSICS,
+              languages = listOf(Language.ENGLISH),
+              tutorUid = listOf(),
+              studentUid = "student123",
+              minPrice = 20.0,
+              maxPrice = 50.0,
+              timeSlot = "16/11/2024T12:00:00",
+              status = LessonStatus.STUDENT_REQUESTED,
+              latitude = 46.518973490411526,
+              longitude = 6.5685102716088295),
+      )
+
+  private val mockStudent =
+      Profile(
+          "student123",
+          "mockTutor",
+          "Ozymandias",
+          "Halifax",
+          "1234567890",
+          Role.STUDENT,
+          Section.IN,
+          AcademicLevel.BA3,
+          listOf(Language.ENGLISH),
+          listOf(Subject.AICC),
+          List(7) { List(12) { 0 } })
+
+  @get:Rule val composeTestRule = createComposeRule()
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
+      GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getProfiles(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+      onSuccess(listOf(mockStudent)) // Simulate a list of profiles with our beloved Ozymandias
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess()
+    }
+    whenever(mockLessonRepository.getLessonsForTutor(eq("mockUid"), any(), any())).thenAnswer {
+        invocation ->
+      val onSuccess = invocation.arguments[1] as (List<Lesson>) -> Unit
+      onSuccess(mockLessons)
+    }
+    whenever(mockLessonRepository.getAllRequestedLessons(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Lesson>) -> Unit
+      onSuccess(mockLessons)
+    }
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    val mockLessonFlow = MutableStateFlow<Lesson?>(mockLessons[0])
+    doReturn(mockLessonFlow).`when`(mockLessonViewModel).selectedLesson
+    mockLessonFlow.value = mockLessons[0]
+  }
+
+  @Test
+  fun TutorEndToEndTest() {
+    composeTestRule.setContent {
+      PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
+    }
+    // Thread.sleep(50000)
```

Comment:
This end to end passes the CI without the sleep being active? That's a great improvement over its student counterpart!